### PR TITLE
LPS-161349 [TESTRAY] Fixed behavior of checkbox check all / uncheck all,  to correctly behavior

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:stable
ci:test:relevant

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/components/ListView/ListView.tsx [hunk 1]
```diff
@@ -169,7 +172,8 @@ const ListViewRest: React.FC<ListViewProps> = ({
 	);
 
 	const onSelectAllRows = useCallback(() => {
-		items.forEach(onSelectRow);
+		const rowIds = items.map(({id}) => id);
+		onSelectRow(rowIds);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/components/ListView/ListView.tsx:176]: Inline

```suggestion
	onSelectRow(items.map(({id}) => id));
```

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/components/ListView/ListView.tsx [hunk 2]
```diff
@@ -178,6 +182,18 @@ const ListViewRest: React.FC<ListViewProps> = ({
...
+		if (tableProps.rowSelectable) {
+			const initialChecked = items.every(({id}) =>
+				selectedRows.includes(id)
+			);
+			dispatch({
+				payload: initialChecked,
+				type: ListViewTypes.SET_CHECKED_ALL_ROWS,
+			});
+		}
+	}, [items, tableProps, selectedRows, dispatch]);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/components/ListView/ListView.tsx:195]: Add space

```suggestion
	useEffect(() => {
		if (tableProps.rowSelectable) {
			const initialChecked = items.every(({id}) =>
				selectedRows.includes(id)
			);
			
			dispatch({
				payload: initialChecked,
				type: ListViewTypes.SET_CHECKED_ALL_ROWS,
			});
		}
	}, [items, tableProps, selectedRows, dispatch]);
```

or inline

```suggestion
	useEffect(() => {
		if (tableProps.rowSelectable) {
			dispatch({
				payload: items.every(({id}) =>
				     selectedRows.includes(id)
			        ),
				type: ListViewTypes.SET_CHECKED_ALL_ROWS,
			});
		}
	}, [items, tableProps, selectedRows, dispatch]);
```

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/context/ListViewContext.tsx [hunk 3]
```diff
@@ -91,22 +95,39 @@ export const ListViewContext = createContext<
...
+				const rowAlreadyInserted = state.selectedRows.includes(
+					rowIds as number
+				);
+
+				if (rowAlreadyInserted) {
+					selectedRows = selectedRows.filter((row) => row !== rowIds);
+				}
+				else {
+					selectedRows = [...selectedRows, rowIds as number];
+				}
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/context/ListViewContext.tsx:117]: Simplify

```suggestion
		selectedRows = rowAlreadyInserted ? 
		    selectedRows.filter((row) => row !== rowIds): 
		    [...selectedRows, rowIds as number];			
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
Good job, @welitonpe take a look im my suggestions, please 🗡️

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**